src/main/java/InfrastructureManager/Master.java

```diff
@@ -175,9 +226,13 @@ public static Master getInstance() {
         return instance;
     }
 
+
+
     public static void main(String[] args) {
         Master.getInstance().startMainRunner();
+        Master.getInstance().startRunnerThread("RestInputTest");
         Master.getInstance().startRunnerThread("RestServer");
+        Master.getInstance().startRunnerThread("MMInputTest");
```

Comment:
no longer needed

Reply:
I believe it is needed because that is the thread that has MatchMaker as input, so is the one that serves the clients with the REST resource. I could maybe change the names to make that more clear (get rid of "test")

Reply:
Now that I think about it, even RestServer is not needed here, you can always start them when you are need them, but it should be possible to not start them if the use case doesn't include a rest server or match-making.